src/V3Config.cpp

```diff
@@ -145,9 +146,289 @@ class V3ConfigIgnores {
 
 V3ConfigIgnores V3ConfigIgnores::s_singleton;
 
+//######################################################################
+// variable attributes (public, clock, isolate_assignments, ...)
+
+class V3ConfigVarAttr {
+    typedef std::tuple<AstAttrType,FileLine*,AstSenTree*> AttrEntry;
+    typedef std::vector<AttrEntry> AttrEntryVector;
+    typedef std::map<string,AttrEntryVector> VarAttrMap;
+    typedef std::map<string,VarAttrMap> VarAttrMapFTask;
+    typedef std::map<string,VarAttrMapFTask> AttrMap;
+    typedef std::unordered_set<std::string> FunctionList;
+
+    AttrMap m_varAttrs;
+
+    std::map<std::string,FunctionList> m_functionIsolate;
+
+    static V3ConfigVarAttr s_singleton;  // Singleton (not via local static, as that's slow)
+    V3ConfigVarAttr() {}
+    ~V3ConfigVarAttr() {}
+
+    void applyVarAttr(AstVar* varp, AttrEntry entry) {
+        AstNode* newp = new AstAttrOf(varp->fileline(), std::get<0>(entry));
+        varp->addAttrsp(newp);
+        if (std::get<0>(entry) == AstAttrType::VAR_PUBLIC_FLAT_RW) {
+            newp->addNext(new AstAlwaysPublic(varp->fileline(), std::get<2>(entry), NULL));
+        }
+    }
+    void applyVarAttrForTask(const VarAttrMap& map, AstVar* varp) {
+        const string var = varp->name();
+        for (VarAttrMap::const_iterator it = map.begin(); it != map.end(); ++it) {
+            if (VString::wildmatch(var.c_str(), it->first.c_str())) {
+                for (AttrEntryVector::const_iterator eit = it->second.begin();
+                     eit != it->second.end(); ++eit) {
+                    applyVarAttr(varp, *eit);
+                }
+            }
+        }
+    }
+    void applyVarAttrForModule(const VarAttrMapFTask& map, const string& ftask, AstVar* varp) {
+        for (VarAttrMapFTask::const_iterator it = map.begin(); it != map.end(); ++it) {
+            if (VString::wildmatch(ftask.c_str(), it->first.c_str())) {
+                applyVarAttrForTask(it->second, varp);
+            }
+        }
+    }
+    void applyFunctionIsolate(const FunctionList& list, AstNodeFTask* funcp) {
+        for (FunctionList::const_iterator it = list.begin(); it != list.end(); ++it) {
+            if (VString::wildmatch(funcp->name().c_str(), it->c_str())) {
+                funcp->attrIsolateAssign(true);
+            }
+        }
+    }
+public:
+    inline static V3ConfigVarAttr& singleton() { return s_singleton; }
+    void addVarAttr(FileLine* fl, const string& module, const string& ftask, const string& var,
+                    AstAttrType type, AstSenTree* sensep) {
+        m_varAttrs[module][ftask][var].push_back(std::make_tuple(type, fl, sensep));
+    }
+    void addFunctionIsolate(const string& module, const string& function) {
+        m_functionIsolate[module].insert(function);
+    }
+    void applyVarAttr(AstNodeModule* modulep, AstVar* varp) {
+        const string module = modulep->name();
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [Variable 'module' is assigned a value that is never used.](https://app.codacy.com/manual/wsnyder/verilator/pullRequest?prid=4742436)